packages/grafana-flamegraph/src/FlameGraphHeader.tsx

```diff
@@ -0,0 +1,299 @@
+
+type Props = {
+  search: string;
+  setSearch: (search: string) => void;
+  selectedView: SelectedView;
+  setSelectedView: (view: SelectedView) => void;
+  containerWidth: number;
+  onReset: () => void;
+  textAlign: TextAlign;
+  onTextAlignChange: (align: TextAlign) => void;
+  showResetButton: boolean;
+  colorScheme: ColorScheme | ColorSchemeDiff;
+  onColorSchemeChange: (colorScheme: ColorScheme | ColorSchemeDiff) => void;
+  stickyHeader: boolean;
+  vertical?: boolean;
+  isDiffMode: boolean;
+  getTheme: () => GrafanaTheme2;
+
+  extraHeaderElements?: React.ReactNode;
+};
+
+const FlameGraphHeader = ({
+  search,
+  setSearch,
+  selectedView,
+  setSelectedView,
+  containerWidth,
+  onReset,
+  textAlign,
+  onTextAlignChange,
+  showResetButton,
+  colorScheme,
+  onColorSchemeChange,
+  stickyHeader,
+  extraHeaderElements,
+  vertical,
+  isDiffMode,
+  getTheme,
+}: Props) => {
+  const styles = getStyles(getTheme(), stickyHeader);
+  const [localSearch, setLocalSearch] = useSearchInput(search, setSearch);
+
+  const suffix =
+    localSearch !== '' ? (
+      <Button
+        icon="times"
+        fill="text"
+        size="sm"
+        onClick={() => {
+          // We could set only one and wait them to sync but there is no need to debounce this.
+          setSearch('');
+          setLocalSearch('');
+        }}
+      >
+        Clear
+      </Button>
+    ) : null;
+
+  return (
+    <div className={styles.header}>
+      <div className={styles.inputContainer}>
+        <Input
+          value={localSearch || ''}
+          onChange={(v) => {
+            setLocalSearch(v.currentTarget.value);
+          }}
+          placeholder={'Search..'}
+          suffix={suffix}
+        />
+      </div>
+
+      <div className={styles.rightContainer}>
+        {showResetButton && (
+          <Button
+            variant={'secondary'}
+            fill={'outline'}
+            size={'sm'}
+            icon={'history-alt'}
+            tooltip={'Reset focus and sandwich state'}
+            onClick={() => {
+              onReset();
+            }}
+            className={styles.buttonSpacing}
+            aria-label={'Reset focus and sandwich state'}
+          />
+        )}
+        <ColorSchemeButton
+          value={colorScheme}
+          onChange={onColorSchemeChange}
+          isDiffMode={isDiffMode}
+          getTheme={getTheme}
+        />
+        <RadioButtonGroup<TextAlign>
+          size="sm"
+          disabled={selectedView === SelectedView.TopTable}
+          options={alignOptions}
+          value={textAlign}
+          onChange={onTextAlignChange}
+          className={styles.buttonSpacing}
+        />
+        <RadioButtonGroup<SelectedView>
+          size="sm"
+          options={getViewOptions(containerWidth, vertical)}
+          value={selectedView}
+          onChange={setSelectedView}
+        />
+        {extraHeaderElements && <div className={styles.extraElements}>{extraHeaderElements}</div>}
```

Comment:
Extra elements rendered here. In pyroscope used to inject a button to download the profile in various formats.